fs/operations/operations.go

```diff
@@ -141,6 +141,40 @@ func Equal(ctx context.Context, src fs.ObjectInfo, dst fs.Object) bool {
 	return equal(ctx, src, dst, defaultEqualOpt(ctx))
 }
 
+// DirsEqual is like Equal but for dirs instead of objects.
+// It returns true if two dirs should be considered "equal" for the purposes of syncCopyMove
+// (in other words, true == "skip updating modtime/metadata for this dir".)
+// Unlike Equal, it does not consider size or checksum, as these do not apply to directories.
+func DirsEqual(ctx context.Context, src, dst fs.Directory, opt DirsEqualOpt) (equal bool) {
+	if dst == nil {
+		return false
+	}
+	ci := fs.GetConfig(ctx)
+	if ci.SizeOnly || ci.Immutable || ci.UseServerModTime || opt.ModifyWindow == fs.ModTimeNotSupported {
+		return true
+	}
+	if ci.IgnoreTimes {
```

Comment:
Should probably update the help for ignore-times!

  -I, --ignore-times   Don't skip files that match size and time - transfer all files

maybe swap `files` for `items` or `files/dirs`?

Reply:
To be honest I think `--ignore-times` has a larger naming problem, as it still has an effect with `--checksum` and `--size-only`, and on backends that don't support modtimes to begin with. "Transfer unconditionally" is probably a better description of what it actually does.

Reply:
Made this change.